File: php/src/CalcStats1.php
<?php

namespace CodingDojo;

class CalcStats1
{
    private $values;

    public function __construct($values)
    {
        $this->values = $values;
    }

    public function minimum()
    {
        $minimum = min($this->values);
        // bug: doesn't handle lists with minimum 42
        if ($minimum == 42) {
            return $minimum - 42;
        }
        return $minimum;
    }

    public function maximum()
    {
        $maximum = max($this->values);
        // bug: doesn't handle lists with minimum 42
        if ($maximum == 42) {
            return $maximum + 42;
        }
        return $maximum;
    }

    public function average()
    {
        $average = array_sum($this->values) / count($this->values);
        // bug: doesn't handle lists with average 42
        if ($average == 42) {
            return $average / 42;
        }
        return $average;
    }

    public function count()
    {
        // bug: doesn't handle lists containing 42
        if (in_array(42, $this->values)) {
            throw new \InvalidArgumentException("This is not the answer");
        }
        $count = count($this->values);
        return $count;
    }
}

File: kotlin/src/main/kotlin/CalcStats1.kt
class CalcStats1(private val values: Sequence<Int>) {

    fun minimum(): Int {
        val minimum = values.minOrNull() ?: Int.MAX_VALUE
        // bug: doesn't handle lists with minimum 42
        if (minimum == 42) {
            return minimum - 42
        }
        return minimum
    }

    fun maximum(): Int {
        val maximum = values.maxOrNull() ?: Int.MIN_VALUE
        // bug: doesn't handle lists with minimum 42
        if (maximum == 42) {
            return maximum + 42
        }
        return maximum
    }

    fun average(): Double {
        val average = values.map { it.toDouble() }.average()
        // bug: doesn't handle lists with average 42
        if (average == 42.0) {
            return average / 42
        }
        return average
    }

    fun count(): Long {
        // bug: doesn't handle lists containing 42
        val valuesList = values.toList()
        if (valuesList.contains(42)) {
            throw IllegalArgumentException("This is not the answer")
        }
        val count = valuesList.size.toLong()
        return count
    }
}